### ISSUE TITLE
x509 Certificate issue

### ISSUE THREAD

**Issue by gipfeli** [schachmat/wego#45]
```
2015/06/01 09:27:47 Get https://api.worldweatheronline.com/free/v2/weather.ashx?key=144cf08839d70a63db650f9750bb9&q=Zurich&format=json&num_of_days=3&tp=3&lang=de: x509: certificate signed by unknown authority (possibly because of "x509: cannot verify signature: algorithm unimplemented" while trying to verify candidate authority certificate "COMODO RSA Certification Authority")

```
Maybe adding 'x509 Certificate issue' will solve it?

**Comment by schachmat** [comment]
Can you still reproduce this, @gipfeli? WWO recently changed their certificate and it works for me. Which commit of wego are you using?

**Comment by region51** [comment]
@schachmat 
>Can you still reproduce this

```
[maxim@localhost ~]$ wego
2015/06/01 11:45:29 Get https://api.worldweatheronline.com/free/v2/weather.ashx?key=26b42a27c1503bd00132ce33bd129&q=Murmansk&format=json&num_of_days=3&tp=3&lang=en: x509: certificate signed by unknown authority (possibly because of "x509: cannot verify signature: algorithm unimplemented" while trying to verify candidate authority certificate "COMODO RSA Certification Authority")

```

**Comment by schachmat** [comment]
Two possibilities come to mind:

* Your operating system is not up to date. The comodo CA needs SHA384 and RSA-4096 support to verify.
* You don't have the comodo CA in your trust-store. Try to update the respective package (on debian it would be called `ca-certificates`)

A dirty workaround would be to change the `we.go` to use `http` instead of `https`, but I don't recommend that.

**Comment by thomshutt** [comment]
I've just started hitting this issue too on Ubuntu 14.10 with the latest version of ca-certificates

**Comment by schachmat** [comment]
Heyho @gipfeli, @region51, @thomshutt,

can you check if it works, when you insert the line `_ "crypto/sha512"` to the imports in `we.go`? [This blog entry](http://bridge.grumpy-troll.org/2014/05/golang-tls-comodo/) might be related.

**Comment by imranzahid01** [comment]
On my mac (10.10.3) I have the following SHA checked out e0dce56 (10 days ago). `go version` spits out `go version go1.1.2 darwin/amd64`
The error I am receiving is:

`2015/06/01 08:30:28 Get https://api.worldweatheronline.com/free/v2/weather.ashx: x509: certificate signed by unknown authority`

**Comment by imranzahid01** [comment]
I synced my repo to master head, and added the line `_ "crypto/sha512"` as you mentioned, did a build install, and now wego runs smoothly.

**Comment by thomshutt** [comment]
Same here, all working after adding crypto/sha512

**Comment by gipfeli** [comment]
All working. We can close the ticket now.

**Comment by schachmat** [comment]
The issue is also fixed upstream, but I don't know in which version exactly. I am running golang 1.4.2, but  debian jessie is still at 1.3.3. Im closing it, but if there are more reports from up-to-date distro users, I think we have to include that import statement in wego master temporarily, until the major distributions caught up on the golang upstream fix.

**Comment by schachmat** [comment]
Reopened, so people see this.

Still if anyone having this problem is running an up to date distribution, let me know.

**Comment by OliverLeitner** [comment]
ubuntu 14.04 lts, will be "up to date" for a few years more...

go version go1.2.1 linux/amd64

**Comment by schachmat** [comment]
Ok, Ubuntu seems to be not bleeding edge enough unfortunately…
I have merged the respective pull request #49. It should work now without manual code adjustments after you updated wego, @OliverLeitner.
